### PR TITLE
--enable-distro implies --enable-reproducible-build

### PR DESCRIPTION
**Pull request by douzzer**
turn on reproducible-build by default when enable-distro.

### CODE REVIEW

**Review by dgarske — APPROVED** [review]
Thanks @douzzer . I also discussed this with @lechner prior to this, so very happy with this change.

**Comment by lechner** [comment]
@dgarske While the news are a bit lame after the merge, [this patch](https://salsa.debian.org/lechner/wolfssl/-/blob/debian/master/debian/patches/9dadd02fb9435cccd170765bbd91fb74e0cbc71d.diff) passed [Salsa CI](https://salsa.debian.org/lechner/wolfssl/-/pipelines/233601), a good set of QA checks in Debian. This PR was requested by the Debian maintainer. Thanks @douzzer !

**Pull request merged**
